nixos/modules/programs/fish.nix

```
@@ -29,6 +45,14 @@ in
         type = types.bool;
       };
 
+      translateEnvironment = mkOption {
```

The environment is "translated" either way, it's just what is used to translate it changes. Alternatively, change the type to `types.enum [ "babelfish" "foreign-env" ]`.

```suggestion
      useBabelfish = mkOption {
```

---

I can see that. I'll use the enum approach and default to "foreign-env".

---

Actually, an enum with two values is pretty weird to switch on (if-else if-else? but the type should make the else useless). I think I'll go with `useBabelfish` unless a third option is added to the module.

---

Maybe an attrset is a nice solution?
```
{ "foreign-env" = fenv-case; "babelfish" = babelfish-case; }.${cfg.translator}
```